codec-http/src/main/java/io/netty/handler/codec/http/HttpContentCompressor.java

```diff
@@ -107,6 +136,10 @@ protected Result beginEncode(HttpResponse headers, String acceptEncoding) throws
             return null;
         }
 
+        if (!useCompression(headers)) {
```

Comment:
@teaey sorry one last nit... I think this should be the first thing to check as we not need to even compare headers if it is false. Please move it before the header check